whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceTestIntegration.java

```diff
@@ -5026,6 +5026,538 @@ public void primary_key_comment() {
+                "auth:        SSO person@net.net\n" +
+                "mnt-by:      ANOTHER-MNT\n" +
+                "source:      TEST"));
+
+        final WhoisResources person = RestTest.target(getPort(), "whois/test/person?password=test")
+                .request()
+                .post(Entity.entity(map(createPerson), MediaType.APPLICATION_XML), WhoisResources.class);
+
+        assertThat(person.getWhoisObjects().get(0).getAttributes().get(5),
+                is(new Attribute("mnt-ref", "OWNER-MNT", null, "mntner", new Link("locator", "http://rest-test.db.ripe" +
+                        ".net/test/mntner/OWNER-MNT"), null)));
+
+        final RpslObject updatePerson = RpslObject.parse("" +
+                "person:    Pauleth Palthen\n" +
+                "address:   Singel 258\n" +
+                "phone:     +31-1234567890\n" +
+                "e-mail:    noreply@ripe.net\n" +
+                "mnt-by:    OWNER-MNT\n" +
+                "mnt-ref:    ANOTHER-MNT\n" +
+                "nic-hdl:   PP1-TEST\n" +
+                "remarks:   remark\n" +
+                "source:    TEST\n");
+
+        final WhoisResources updateResponse = RestTest.target(getPort(), "whois/test/person/PP1-TEST?password=test")
+                .request()
+                .put(Entity.entity(map(updatePerson), MediaType.APPLICATION_XML), WhoisResources.class);
+
+        assertThat(updateResponse.getWhoisObjects().get(0).getAttributes().get(5),
+                is(new Attribute("mnt-ref", "ANOTHER-MNT", null, "mntner", new Link("locator", "http://rest-test.db.ripe" +
+                        ".net/test/mntner/ANOTHER-MNT"), null)));
+    }
+
+    @Test
+    public void create_role_valid_person_mnt_ref() {
+
+        databaseHelper.addObject(RpslObject.parse("" +
+                "mntner:      ANOTHER-MNT\n" +
+                "descr:       Owner Maintainer\n" +
+                "admin-c:     TP1-TEST\n" +
+                "upd-to:      noreply@ripe.net\n" +
+                "auth:        MD5-PW $1$6NVP0chT$XJgJwKM60eoVnMhIc/4hu/ #another\n" +
+                "auth:        SSO person@net.net\n" +
+                "mnt-by:      ANOTHER-MNT\n" +
+                "source:      TEST"));
+
+        databaseHelper.addObject(RpslObject.parse("" +
+                "person:    Pauleth Palthen\n" +
+                "address:   Singel 258\n" +
+                "phone:     +31-1234567890\n" +
+                "e-mail:    noreply@ripe.net\n" +
+                "mnt-by:    OWNER-MNT\n" +
+                "mnt-ref:   ANOTHER-MNT\n" +
+                "nic-hdl:   PP1-TEST # create comment\n" +
+                "remarks:   remark\n" +
+                "source:    TEST\n"));
+
+        final RpslObject roleObject = RpslObject.parse("" +
+                "role:          SWE test\n" +
+                "address:       123\n" +
+                "e-mail:        123@ripe.net\n" +
+                "nic-hdl:       ROLE-TEST\n" +
+                "tech-c:        PP1-TEST\n" +
```

Comment:
Test case says create role with mnt_ref. But there is no mnt_ref in role object? instead assertion is for mnt_by?